ui/src/app/images/views/ImageList/UbuntuImages/CustomSource/CustomSource.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState } from "react";
+
+import CustomSourceConnect from "./CustomSourceConnect";
+import CustomSourceImages from "./CustomSourceImages";
+
+const CustomSource = (): JSX.Element => {
+  const [connected, setConnected] = useState(false);
+  const [sourceUrl, setSourceUrl] = useState("");
```

Comment:
I think `null` is probably a better value for when there is no source, you'll have to update the types to `string | null`.